src/dialect/mod.rs

```diff
@@ -49,6 +49,39 @@ pub trait Dialect: Debug + Any {
     fn is_identifier_start(&self, ch: char) -> bool;
     /// Determine if a character is a valid unquoted identifier character
     fn is_identifier_part(&self, ch: char) -> bool;
+
+    /// The name of the dialect
+    fn dialect_name(&self) -> &'static str;
+
+    /// Enable the parser to implement dialect specific functionality.
+    /// The input for this function a list dialect names.
+    /// Function will return true if the current dialect is a subset of the input.
+    ///
+    /// parser usage exmple:
+    /// `if self.dialect.is_dialect(vec!["mssql"]) {
+    ///    // some special mssql behaviour  
+    /// } else {
+    ///   // defualt bahviour
+    /// }`
+    fn is_dialect(&self, dialects: Vec<&str>) -> bool {
+        dialects.contains(&self.dialect_name())
```

Comment:
In a recently merged PR, an other way to switch on certain dialects has been added:
https://github.com/ballista-compute/sqlparser-rs/pull/254

Reply:
thanks, bad rebase issue , fixed